js/src/framework.ts

```diff
@@ -210,10 +255,43 @@ export async function runEvaluator(
   if (typeof evaluator.data === "string") {
     throw new Error("Unimplemented: string data paths");
   }
-  const dataResult = evaluator.data();
-  let data = null;
+  let dataResult =
+    typeof evaluator.data === "function" ? evaluator.data() : evaluator.data;
+
+  if ("_type" in dataResult) {
+    if (dataResult._type !== "BaseExperiment") {
+      // For some reason, the typesystem won't let me check if dataResult._type === "BaseExperiment"
```

Comment:
Not sure I follow. The TS compiler already thinks it's a `BaseExperiment` at this point, so this is just an extra-cautious check that they didn't put something undesirable in `_type` somehow. So why would we rather check `dataResult._type === "BaseExperiment"`?

Reply:
I had originally written `if ("_type" in dataResult && dataResult._type === "BaseExperiment")` which wasn't enough to convince the compiler that we've covered all cases where `dataResult` can be a `BaseExperiment`